src/Components/Firestore.js

```diff
@@ -303,3 +319,69 @@ export async function MarkNotificationsSeenOrRead(notiArray, IdToNotify, verb) {
     }
   }
 }
+
+export async function getRandomCommunityList() {
+  // Repeat call to try and ensure a list with at least (1) anime title is found.
+  for (let i = 1; i < 6; i++) {
+    try {
+      const randomNumber = Math.floor(Math.random() * 9999999);
```

Comment:
maybe make the random number generator into a private function in this file -- the implementation has to be exactly the same here and above for this method to work, so it might make sense to just have one implementation

also, I think 10 million (technically, 9,999,999) is a safe number -- hopefully we should have an algorithmic watchlist recommendation system well before we get anywhere close to saturating that keyspace.

Reply:
Could you elaborate on the benefit of using a private function here?  I'm not understanding how this would generate the random # any differently.

Reply:
it wouldn't generate the number any differently, but it would guarantee the number generation happening during read and write are done the same way.  I imagine if you used different numbers in each location, it would break the system.  e.g. if you used 9999999 during write, but 99999999 during read (can you spot the difference?), you'd get the same watchlist like 90% of the time.  

instead, either consider a const for the number, like:

```js

const watchlistRandomMax = 9999999;

// then later...

// on document creation

const random = Math.floor(Math.random() * watchlistRandomMax);

const data = {
  ownerId: "whoever",
  listId: "whatever",
  random: random
};
setDoc(db, watchlistRef, data);

// and on read, as well:

const random = Math.floor(Math.random() * watchlistRandomMax);

```

or else the whole generation in a function, like:

```js

function generateRandomWatchlistInt() {
  return Math.floor(Math.random() * 9999999);
}

// then, later...

// on document creation

const data = {
  ownerId: "whoever",
  listId: "whatever",
  random: generateRandomWatchlistInt()
};

await setDoc(db, watchlistRef, data);

// and on document reads...

const random = generateRandomWatchlistInt()

```

this makes the consistency between these implementations explicit, instead of implicit.

but an optional thing, up to you if you want to apply this. making the number const is quick and easy and probably good practice, just to make sure it is the same.

Reply:
Okay, I see. Yes, that's a very good point! Thanks for the explanation!!!

Will incorporate prior to merging.